### PR TITLE
DDF-4811 Match the coordinate order of each WFS 1.1.0 query to that of its source

### PR DESCRIPTION
**Pull request by jrnorth**
#### What does this PR do?
Updates the WFS 1.1.0 `FilterDelegate` to use the configured coordinate order of its corresponding source to determine the coordinate order to use when it constructs the query.

#### Who is reviewing it?
@Corey-Collins 
@samuelechu 
@leo-sakh 
@glenhein 

#### Select relevant component teams: 
@codice/io 
@codice/ogc 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@jlcsmith

#### How should this be tested?
1. Run `log:set info org.apache.cxf` and `log:set warn org.apache.camel`.
2. Create a WFS 1.1.0 source with the URL http://services.azgs.az.gov/ArcGIS/services/aasggeothermal/AZWellLogs/MapServer/WFSServer
3. Run various `anyGeo` queries against the source, using `Line`, `Polygon`, `Point-Radius`, and `Bounding Box`.
a. When you run a query there will be a `GetFeature` XML blob printed in the logs that contains the geospatial query. Use this to verify each query's correctness.
b. For each query, run it as both `Lat/Lon` and `Lon/Lat` and verify the outgoing `GetFeature` XML has the right coordinate order in both cases. Check the `Coordinate Order` in your WFS source's configuration, then run your query and verify that the coordinate order in the `GetFeature` matches. Change the `Coordinate Order` in the source configuration, rerun the same query, and verify the coordinate order in the `GetFeature` changed appropriately.

#### What are the relevant tickets?
Fixes: #4811 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsFilterDelegate.java [hunk 1]
```diff
@@ -1078,12 +1082,20 @@ private FilterType buildSingleGeospatialFilter(
 
     Envelope envelope = createEnvelopeFromWkt(wkt);
 
+    List<Double> lowerCornerCoordinates = asList(envelope.getMinX(), envelope.getMinY());
```
Comment (glenhein) [inline catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsFilterDelegate.java]: ✏️ In an ideal world, I would like a more pure OO solution. For example, a class called LatLonList with a ctor that takes two doubles and the isLatLonOrder flag. The class would then have an asList() method that returns a java.util.List that is constructed in the correct order. The objective is to group all the data (the double values and flag) with the methods (the asList) into one class.

File: catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsFilterDelegate.java [hunk 2]
```diff
@@ -1097,14 +1109,11 @@ private FilterType buildSingleGeospatialFilter(
 
     Coordinate[] coordinates = getCoordinatesFromWkt(wkt);
     if (coordinates != null && coordinates.length > 0) {
-      StringBuilder coordString = new StringBuilder();
-
-      for (Coordinate coordinate : coordinates) {
-        coordString.append(coordinate.y).append(",").append(coordinate.x).append(" ");
-      }
+      String coordinateString =
```
Comment (glenhein) [inline catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsFilterDelegate.java]: ✏️ Similar to my above comment, create a class that encapsulates the data and methods.

File: catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsFilterDelegate.java (hunk 1 above)
Comment (jrnorth) [reply catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsFilterDelegate.java]: I've updated this class to use the strategy pattern. Now `WfsSource` will pass in either a `LatLonCoordinateStrategy` or `LonLatCoordinateStrategy` and the filter delegate does not have any conditional logic for handling the coordinate order.

File: catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsSource.java [hunk 3]
```diff
@@ -1129,6 +1130,12 @@ public String getCoordinateOrder() {
     return this.coordinateOrder;
   }
 
+  private CoordinateStrategy getCoordinateStrategy() {
+    return LAT_LON_ORDER.equals(coordinateOrder)
```
Comment (samuelechu) [inline catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsSource.java]: ✏️ coordinateOrder.equals(LAT_LON_ORDER) ?

File: catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/test/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/LatLonCoordinateStrategyTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Envelope;
+import org.junit.Test;
+
+public class LatLonCoordinateStrategyTest {
+  private final LatLonCoordinateStrategy latLonCoordinateStrategy = new LatLonCoordinateStrategy();
+
+  @Test
+  public void toStringSingleCoordinateReturnsCommaSeparatedCoordinateValuesInLatLonOrder() {
+    assertThat(latLonCoordinateStrategy.toString(new Coordinate(10, 20)), is("20.0,10.0"));
```
Comment (samuelechu) [inline catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/test/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/LatLonCoordinateStrategyTest.java]: Is this type of test necessary? It seems kind of brittle

Comment (jrnorth) [reply catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/test/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/LatLonCoordinateStrategyTest.java]: Brittle in what way? The coordinate strings must be formatted in a certain way for geospatial queries to work.

Comment (samuelechu) [reply catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/test/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/LatLonCoordinateStrategyTest.java]: Brittle in that when the format changes, we would have to update all the tests that check for a certain string being returned. Not sure how often the format would change though

Comment (jrnorth) [reply catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/test/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/LatLonCoordinateStrategyTest.java]: We need to make sure this is formatted correctly so the query will work. It's not going to change often.

**Comment by millerw8** [comment]
build now

**Comment by jrnorth** [comment]
build now

**Comment by bdeining** [comment]
verified coordinate order and various geo queries based on the cxf logs ✅

**Pull request merged**